packages/moonstone/Slider/SliderBehaviorDecorator.js

```diff
@@ -77,14 +77,19 @@ const SliderBehaviorDecorator = hoc(defaultConfig, (config, Wrapped) => {
 				active: false,
 				dragging: false,
 				focused: false,
-				useHintText: false
+				useHintText: false,
+				valueClone: props.value
 			};
 		}
 
-		UNSAFE_componentWillReceiveProps (nextProps) {
-			if (this.props.value !== nextProps.value) {
-				this.setState({useHintText: false});
+		static getDerivedStateFromProps (nextProps, prevState) {
+			if (nextProps.value !== prevState.valueClone) {
```

Comment:
We don't have a specific coding rule for naming in `getDerivedStateFromProps`, but let's follow `react`'s examples. Use `(props, state)` instead of `(nextProps, prevState)`, and `prev*` instead of `*Clone`
e.g.
```
		static getDerivedStateFromProps (props, state) {
			if (props.value !== state.prevValue) {
				return ({
					useHintText: false,
					prevValue: props.value
				});
			}
```